starcounter-include.html

```diff
@@ -210,35 +208,67 @@
+         */
+        StarcounterIncludePrototype._renderCompositionChange = function(compositionRef, callb) {
+            if (!this._forceLayoutChange && this._lastCompositionRef == compositionRef) {
+                return;
+            }
+            this.storedLayout = null;
+            let composition = callb();
+            this.stampComposition(composition);
+            this._forceLayoutChange = false;
+            this._lastCompositionRef = compositionRef;
+        }
+
         /**
          * Handles change of the composition.
          * If not given explicitely, fetches one from provider.
-         * Stamps if needed.
+         * If not given in provider, fetches one from imported template's default composition.
+         * If imported template does not have a default composition, uses fallback composition.
+         * Warning: the explicit composition might come from https://github.com/Starcounter/starcounter-layout-html-editor/blob/17b21f729facd9a8dcd4241fb5c48cb71de11af5/starcounter-layout-html-editor.html#L253
          * @see .stampComposition
          * @param  {String} compositionString stringified HTML for new Shadow Root
          */
         StarcounterIncludePrototype._compositionChanged = function(compositionString) {
-            let composition;
-            if (this.partial && compositionString === undefined) {
-                const compositionProvider = findCompositionProvider(this.partial);
-                if (compositionProvider) {
-                    compositionString = compositionProvider.Composition$; //should always be string
-                    this.storedLayout = compositionString;
-                }
-            }
-            // do not change if there is one, it's same as before, and we are not forced
-            if (!compositionString) {
-                composition = this.defaultComposition && this.defaultComposition.cloneNode(true);
-                compositionString = this.defaultComposition;
-            } else {
-                // make clone to avoid direct binding
-                composition = this.stringToDocumentFragment(compositionString);
-            }
-            if (!compositionString || compositionString !== this._lastLayout || this._forceLayoutChange) {
-                this.stampComposition(composition);
-                this._forceLayoutChange = false;
-                this._lastLayout = compositionString;
+            var that = this;
+
+            if (compositionString) {
+                this._forceLayoutChange = true;
+                this._renderCompositionChange("explicit or fallback", function() {
+                    return that.stringToDocumentFragment(compositionString);
+                });
             }
+            else {
+                const compositionProvider = this.partial && findCompositionProvider(this.partial);
+                this.partialId = compositionProvider && compositionProvider.PartialId;
```

Comment:
I would check for `compositionProvider` in an `if` before L248 and save all these `&&`'s.